resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/BaseYarnClusterSuite.scala

```diff
@@ -161,6 +162,11 @@ abstract class BaseYarnClusterSuite
     }
     extraJars.foreach(launcher.addJar)
 
+    if (outFile.isDefined) {
```

Comment:
`outFile.foreach`?

Reply:
I think that `foreach` on an option is unclear (looks like a loop) and should be avoided unless it really simplifies the logic, which it doesn't do here.

Reply:
If you do a foreach then the `.get` goes away and the code could be a little cleaner, but it's pretty minor.

Reply:
Like I said, I think `foreach` is a bad practice with options, so I'd rather not change to use it. I'd be happy to change this to a pattern match if you think it is really desirable to get rid of the `.get`.

Reply:
To me, either way is fine.